lib/kitchen/driver/oci.rb

```diff
@@ -265,60 +385,100 @@ def user_data
-        OCI::Core::Models::LaunchInstanceDetails.new.tap do |l|
-          hostname = generate_hostname
+      def dbaas_launch_details # rubocop:disable Metrics/MethodLength
+        cpu_core_count = config[:dbaas][:cpu_core_count] ||= 2
+        database_edition = config[:dbaas][:database_edition] ||= OCI::Database::Models::DbSystem::DATABASE_EDITION_ENTERPRISE_EDITION
+        initial_data_storage_size_in_gb = config[:dbaas][:initial_data_storage_size_in_gb] ||= 256
+        license_model = config[:dbaas][:license_model] ||= OCI::Database::Models::DbSystem::LICENSE_MODEL_BRING_YOUR_OWN_LICENSE
+
+        OCI::Database::Models::LaunchDbSystemDetails.new.tap do |l|
           l.availability_domain = config[:availability_domain]
           l.compartment_id = config[:compartment_id]
-          l.display_name = hostname
-          l.source_details = instance_source_details
+          l.cpu_core_count = cpu_core_count
+          l.database_edition = database_edition
+          l.db_home = create_db_home_details
+          l.display_name = [config[:hostname_prefix], random_string(4), random_number(2)].compact.join('-')
+          l.hostname = generate_hostname
           l.shape = config[:shape]
-          l.create_vnic_details = create_vnic_details(hostname)
+          l.ssh_public_keys = pubkey
+          l.cluster_name = [config[:hostname_prefix].split('-')[0], random_string(3), random_number(2)].compact.join('-')
+          l.initial_data_storage_size_in_gb = initial_data_storage_size_in_gb
+          l.node_count = 1
+          l.license_model = license_model
+          l.subnet_id = config[:subnet_id]
           l.freeform_tags = process_freeform_tags(config[:freeform_tags])
         end
       end
 
-      def compute_instance_request(state)
-        request = base_oci_launch_details
+      def create_db_home_details
+        raise 'db_version cannot be nil!' if config[:dbaas][:db_version].nil?
 
-        inject_powershell(state) if config[:setup_winrm] == true
+        OCI::Database::Models::CreateDbHomeDetails.new.tap do |l|
+          l.database = create_database_details
+          l.db_version = config[:dbaas][:db_version]
+          l.display_name = ['dbhome', random_number(10)].compact.join('')
+        end
+      end
 
-        metadata = {}
-        metadata.store('ssh_authorized_keys', pubkey)
-        data = user_data
-        metadata.store('user_data', data) if config[:user_data].any?
-        request.metadata = metadata
-        request
+      def create_database_details # rubocop:disable Metrics/MethodLength
+        character_set = config[:dbaas][:character_set] ||= 'AL32UTF8'
+        ncharacter_set = config[:dbaas][:ncharacter_set] ||= 'AL16UTF16'
+        db_workload = config[:dbaas][:db_workload] ||= OCI::Database::Models::CreateDatabaseDetails::DB_WORKLOAD_OLTP
+        admin_password = config[:dbaas][:admin_password] ||= random_password
+        db_name = config[:dbaas][:db_name] ||= 'dbaas1'
```

Comment:
Rubocop complains that this assignment is pointless.  Not sure if you meant that db_name should be assigned to l.db_name below?

Reply:
no, i forgot to move the assigned variable down into the CreateDatabaseDetails